types/styled-components/index.d.ts

```diff
@@ -48,7 +47,9 @@ export interface Styles {
 
 export type InterpolationFunction<P> = (props: P) => Interpolation<P>;
 
-type Attrs<P, A extends Partial<P>, T> = {
+type Attrs<P, A extends Partial<P>, T> = ((props: ThemedStyledProps<P, T>) => A) | A;
+
+type DeprecatedAttrs<P, A extends Partial<P>, T> = {
     [K in keyof A]: ((props: ThemedStyledProps<P, T>) => A[K]) | A[K]
 };
```

Comment:
https://github.com/styled-components/styled-components/pull/2200 implements the function style, [this commit](https://github.com/styled-components/styled-components/commit/ba34abe681f4962a5ee160421bb80eede0d196a3) deprecates the object-with-functions style

Reply:
Thanks for jumping on this so quickly!